### DISCUSSION TITLE
SB5.1 - OCI - Ansible Role: helm-deploy, sunbird-deploy and deploy-player, check deployment result

### DISCUSSION THREAD

**Discussion by heungheung** [Sunbird-Ed/Community#325] [Installation] [answered]
For deploy to k8s, we are now using helm-deploy, sunbird-deploy and deploy-player.

Say for example helm-deploy
  https://github.com/project-sunbird/sunbird-devops/blob/release-5.0.0_RC2/kubernetes/ansible/roles/helm-deploy/tasks/main.yml
the way checking a "deployment" will be in line 113

```
kubectl get deployments.apps {{ release_name }} -o json -n {{ namespace }} | jq -r '.spec.template.spec.containers[0].image | split("/")[1]'
```

kubectl get, output as json and use jq to parse, then do a split.

All these are using assumption that the image pull is in docker hub with the image name format
   dockerhub_user/image_name:tag

if it is from a 3rd party container registry, private registry, jfrog artifactory, the format can be different. it can be something like
  registry_address/account_name/image_name:tag
or even something like 
  registry_address/namespace/project_name/image_name:tag

after doing the split, the name can be [2] or even [3]. Is there an alternate way to do the detection OR can we have something more generic to cater for behavior of different container image registry?

**Comment by heungheung** [comment]
To be more specific, if the objective is to get the last part split with /, don't we do something like

```
kubectl get deployments.apps {{ release_name }} -o json -n {{ namespace }}  | jq -r '.spec.template.spec.containers[0].image ' | awk -F/ '{print $NF}'
```
instead of split with / and always get [1]?

**Reply by beepdot** [reply]
Yes, we will change it below -
```
awk -F/ '{print $NF}'
```
This should work on any container registry as this will give us `image:tag` value.

**Comment by heungheung** [comment, marked as answer]
as replied by keshavprasadms 

> Yes, we will change it below -
> 
> ```
> awk -F/ '{print $(NF-1)":"$(NF)}'
> ```
> 
> This should work on any container registry as this will give us `image:tag` value.

repost as "reply" cannot be marked as answer
